libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -186,15 +172,130 @@ Algorithm selectAlgorithm(TangentSpaceMeshInput *input) {
 
     return outAlgo;
 }
-
 #undef IS_INPUT_TYPE
 
+// The paper uses a Z-up world basis, which has been converted to Y-up here
+void frisvadKernel(const float3& N, float3& T, float3& B) {
+    if (N.y < -1.0f + std::numeric_limits<float>::epsilon()) {
+        // Handle the singularity
+        T = float3{-1.0f, 0.0f, 0.0f};
+        B = float3{0.0f, 0.0f, -1.0f};
+        return;
+    }
+    const float a = 1.0f / (1.0f + N.y);
+    const float b = -N.z * N.x * a;
+    T = float3(b, -N.z, 1.0f - N.z * N.z * a);
+    B = float3(1.0f - N.x * N.x * a, -N.x, b);
```

Comment:
use `float3{}` instead of `float3()` syntax, to be consistant with other places (here and below)

Reply:
done